src/cli/cmd/init/mod.rs

```diff
@@ -0,0 +1,220 @@
+mod dev_shell;
+mod handlers;
+mod project;
+mod prompt;
+mod template;
+
+use clap::Parser;
+use color_eyre::eyre::Result;
+use prompt::Prompt;
+use std::{
+    fs::write,
+    io::IsTerminal,
+    path::PathBuf,
+    process::{exit, ExitCode},
+};
+use url::Url;
+
+use crate::cli::cmd::list::FLAKEHUB_WEB_ROOT;
+
+use super::FlakeHubClient;
+
+use self::{
+    dev_shell::DevShell,
+    handlers::{
+        Flake, Go, Handler, Input, Java, JavaScript, Php, Python, Ruby, Rust, System, Tools, Zig,
+    },
+    project::Project,
+    template::TemplateData,
+};
+
+use super::{CommandExecute, FhError};
+
+// A helper struct for creating FlakeHub URLs
+pub(crate) struct FlakeHubUrl;
+
+impl FlakeHubUrl {
+    fn version(org: &str, project: &str, version: &str) -> String {
+        format!("{FLAKEHUB_WEB_ROOT}/f/{org}/{project}/{version}.tar.gz")
```

Comment:
We should eventually replace this with `Url::path_segments_mut` (and not hardcode the `FLAKEHUB_WEB_ROOT`), but for now this is definitely fine.

Reply:
Might as well just do it now